### ISSUE TITLE
zola serve: `<interface>` and `<port>` are optional option arguments

### ISSUE THREAD

**Issue by rootkea** [getzola/zola#1471]
Hello!

```
$ zola serve -p
Building site...
-> Creating 2 pages (0 orphan), 1 sections, and processing 0 images
Done in 23ms.

Listening for changes in blog{config.toml, content, sass, static, templates}
Press Ctrl+C to stop

Web server is available at http://127.0.0.1:1111

^C$ zola serve -i
Building site...
-> Creating 2 pages (0 orphan), 1 sections, and processing 0 images
Done in 26ms.

Listening for changes in blog{config.toml, content, sass, static, templates}
Press Ctrl+C to stop

Web server is available at http://127.0.0.1:1111

^C$
```
shows that the arguments to `-i` and `-p` are optional and can be skipped. But that's not evident from `zola serve --help`

```
$ zola serve --help
zola-serve 
Serve the site. Rebuild and reload on change automatically

USAGE:
    zola serve [FLAGS] [OPTIONS]

FLAGS:
        --drafts     Include drafts when loading the site
    -f, --fast       Only rebuild the minimum on change - useful when working on a specific page/section
    -h, --help       Prints help information
    -O, --open       Open site in the default browser
    -V, --version    Prints version information

OPTIONS:
    -u, --base-url <base_url>        Changes the base_url [default: 127.0.0.1]
    -i, --interface <interface>      Interface to bind on [default: 127.0.0.1]
    -o, --output-dir <output_dir>    Outputs the generated site in the given path
    -p, --port <port>                Which port to use [default: 1111]
```

Is there a way (some clap conf maybe?) to set:
```
OPTIONS:
    ...
    -i, --interface [interface]      Interface to bind on [default: 127.0.0.1]
    ...
    -p, --port [port]                Which port to use [default: 1111]
```
clearly indicating that `interface` and `port` are optional args to `-i` and `-p` respectively since `[ ]` are used to show optional args in POSIX.

**Comment by rootkea** [comment]
Even `base_url` is an optional arg to `serve -u` which is inconsistent with `build -u base_url` where base_url is a required arg of `-u`. Being discussed in #1466

**Comment by Keats** [comment]
Yup, definitely not intended

**Comment by rootkea** [comment]
I assume you mean `interface`, `port` and `base_url `are NOT intended to be optional args (to `-i`, `-p` and `-u `respectively). That they should be required args like `base_url` is in `build -u base_url`.

If yes then I would like to work on this. Seems simpler and I can finally start with `rust` language! :)

**Comment by Keats** [comment]
Yes, if the arg is given the value should be required!

**Comment by rootkea** [comment]
Fixed by #1474.
